### PR TITLE
Update tidb-configuration-file.md

### PR DESCRIPTION
**Pull request by HuGanghui**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Update tidb-configuration-file.md, add tcp-no-delay

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?
ref pingcap/tidb#23440

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5783
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/5783

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@HuGanghui Is master the only affected version?

**Comment by xhebox** [comment]
@TomShawn Yes, there is no cherry-pick so far.

**Comment by bb7133** [comment]
I will try to review it, after I become a reviewer.

**Comment by qiancai** [comment]
Hi @HuGanghui and @djshow832, we are preparing the user docs for TiDB v5.1 and would like to know whether the doc changes of this PR apply to TiDB v5.1 or not. Would you please check and reply? Thanks!

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -353,6 +353,11 @@ Configuration items related to performance.
 - Determines whether to enable `keepalive` in the TCP layer.
 - Default value: `true`
 
+### `tcp-no-delay`
+
+- Determines whether to enable `no delay` in the TCP layer.
```
Comment (TomShawn) [inline tidb-configuration-file.md:358]:
```suggestion
- Determines whether to enable TCP_NODELAY at the TCP layer. After it is enabled, TiDB disables the Nagle algorithm in the TCP/IP protocol and allows sending small data packets to reduce network latency. This is suitable for latency-sensitive applications with a small transmission volume of data.
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**